tutorials/seis_feature.py

```diff
@@ -30,32 +30,57 @@ def DSAR(data, samp_rate, datas, freqs_names, freqs, Nm, N):
 def nDSAR(dsar):
     return dsar/scipy.stats.zscore(dsar)
     
-
-def compute_hibert(tr, env):
+    
+def compute_physical_features(tr, envfilter = True, freq_bands = [[0.1,1],[1,3],[3,10],[10,20],[20,50]], env_filt = [0.01]):
     
     ''' This functions computes features used by Hibert's research group.
-    tr = Trace of the seismogram as an obspy object
-    env = Envelope of the trace
+    The list of features can be found at - https://www.sciencedirect.com/science/article/pii/S0377027316303948
+    and consult this for a full meaning of individual feature. - https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2016GL070709
+    
+    
+    tr = Trace of the raw seismogram, it could be a filtered seismogram as observed in previous studies. 
+    
+    Look at this code for reference - https://github.com/krischer/mess_2019/tree/master/3_wednesday
+    
     '''
     
     
-    NATT = 57
+    
+    from scipy import signal
+    from scipy.fft import fft, fftfreq
+    import numpy as np
+    from scipy.signal import hilbert
+    
+    
+
+  
+    
+    env = compute_envelope(tr.data)
+    
+    if envfilter == True:
+        sos = signal.butter(1, env_filt, 'lp', fs = tr.stats.sampling_rate,  output = 'sos')
+        env  = signal.sosfilt(sos, env)
+    
+    
+    NATT = 62
     all_attr  = np.empty((1,NATT), dtype = float)
     
-    attributes = ['Duration', 'RappMaxMean', 'RappMaxMedian', 'AsDec','KurtoSig', 'KurtoEnv', 'SkewSig', 
+    attributes = ['Window_Length', 'RappMaxMean', 'RappMaxMedian', 'AsDec','KurtoSig', 'KurtoEnv', 'SkewSig', 
                   'SkewEnv', 'CorPeakNumber', 'Energy1/3Cor', 'Energy2/3Cor', 'int_ratio','E_0.1_1','E_1_3',
                   'E_3_10', 'E_10_20','E_20_50','Kurt_0.1_1','Kurt_1_3','Kurt_3_10','Kurt_10_20','Kurt_20_50',
```

Comment:
these have the hard coded frequency name, i forgot to notice that. it may feel less explanatory, but maybe using E_f1_f2, E_f2_f3, ... would work